bioio/writers/ome_zarr_writer_2.py

```diff
@@ -0,0 +1,688 @@
+        chunk_size = (
+            max(chunk_size[0] // 2, 1),
+            max(chunk_size[1] // 2, 1),
+            max(chunk_size[2] // 2, 1),
+            max(chunk_size[3] // 2, 1),
+            max(chunk_size[4] // 2, 1),
+        )
+    return chunk_size
+
+
+def dim_tuple_to_dict(
+    dims: Union[DimTuple, Tuple[float, float, float, float, float]]
+) -> dict:
+    if len(dims) != 5:
+        raise ValueError("dims must be a 5-tuple in TCZYX order")
+    return {"t": dims[0], "c": dims[1], "z": dims[2], "y": dims[3], "x": dims[4]}
+
+
+def resize(
+    image: da.Array, output_shape: Tuple[int, ...], *args: Any, **kwargs: Any
+) -> da.Array:
+    r"""
+    Wrapped copy of "skimage.transform.resize"
+    Resize image to match a certain size.
+
+    Parameters
+    ----------
+    image: :class:`dask.array`
+        The dask array to resize
+    output_shape: tuple
+        The shape of the resize array
+    \*args: list
+        Arguments of skimage.transform.resize
+    \*\*kwargs: dict
+        Keyword arguments of skimage.transform.resize
+
+    Returns
+    -------
+    Resized image.
+    """
+    factors = np.array(output_shape) / np.array(image.shape).astype(float)
+    # Rechunk the input blocks so that the factors achieve an output
+    # blocks size of full numbers.
+    better_chunksize = tuple(
+        np.maximum(1, np.round(np.array(image.chunksize) * factors) / factors).astype(
+            int
+        )
+    )
+    image_prepared = image.rechunk(better_chunksize)
+
+    # If E.g. we resize image from 6675 by 0.5 to 3337, factor is 0.49992509 so each
+    # chunk of size e.g. 1000 will resize to 499. When assumbled into a new array, the
+    # array will now be of size 3331 instead of 3337 because each of 6 chunks was
+    # smaller by 1. When we compute() this, dask will read 6 chunks of 1000 and expect
+    # last chunk to be 337 but instead it will only be 331.
+    # So we use ceil() here (and in resize_block) to round 499.925 up to chunk of 500
+    block_output_shape = tuple(
+        np.ceil(np.array(better_chunksize) * factors).astype(int)
+    )
+
+    # Map overlap
+    def resize_block(image_block: da.Array, block_info: dict) -> da.Array:
+        # if the input block is smaller than a 'regular' chunk (e.g. edge of image)
+        # we need to calculate target size for each chunk...
+        chunk_output_shape = tuple(
+            np.ceil(np.array(image_block.shape) * factors).astype(int)
+        )
+        return skimage.transform.resize(
+            image_block, chunk_output_shape, *args, **kwargs
+        ).astype(image_block.dtype)
+
+    output_slices = tuple(slice(0, d) for d in output_shape)
+    output = da.map_blocks(
+        resize_block, image_prepared, dtype=image.dtype, chunks=block_output_shape
+    )[output_slices]
+    return output.rechunk(image.chunksize).astype(image.dtype)
+
+
+def _pop_metadata_optionals(metadata_dict: dict) -> dict:
+    for ax in metadata_dict["axes"]:
```

Comment:
maybe
`metadata_dict["axes"] = [ax for ax in metadata_dict["axes"] if ax.get("unit") is not None]`

Reply:
if I did that it would change the behavior of this code.  your suggested change is to only include every axis that has a unit.  the point of this code is to go through each "axis" and remove the "unit" key if it happened to have a None value.  This function is called "pop metadata optionals" - it's just trying to delete keys that aren't holding meaningful data and the spec says are optional.